From 4f52c17c1ec2f32132d391e3c600bbeda8a123cb Mon Sep 17 00:00:00 2001
From: Marc Ferland <marc.ferland@sonatest.com>
Date: Tue, 19 Aug 2025 08:40:23 -0400
Subject: [PATCH] Fix build when KAUTH_BUILD_CODEGENERATOR_ONLY=ON

This fails on my side with:

| -- Building PolkitQt6-1 KAuth backend
| CMake Error at /usr/share/ECM/modules/ECMGenerateQDoc.cmake:67 (message):
|   KF6AuthCore is not a target
| Call Stack (most recent call first):
|   src/CMakeLists.txt:140 (ecm_generate_qdoc)

SPDX-FileCopyrightText: Marc Ferland <marc.ferland@sonatest.com>
SPDX-License-Identifier: MIT

Upstream-Status: Submitted [https://invent.kde.org/frameworks/kauth/-/merge_requests/87]

---
 src/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7565628..efed623 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -137,4 +137,6 @@ ecm_qt_install_logging_categories(
     DESTINATION ${KDE_INSTALL_LOGGINGCATEGORIESDIR}
 )
 
-ecm_generate_qdoc(KF6AuthCore kauth.qdocconf)
+if(NOT KAUTH_BUILD_CODEGENERATOR_ONLY)
+    ecm_generate_qdoc(KF6AuthCore kauth.qdocconf)
+endif ()
-- 
2.34.1

